system-parachains/coretime/coretime-kusama/src/coretime.rs

```diff
@@ -0,0 +1,249 @@
+		// `ref_time` = 7889000 + (3 * 25000000) + (1 * 100000000) = 182889000
+		// `proof_size` = 1636
+		// Add 5% to each component and round to 2 significant figures.
+		let call_weight = Weight::from_parts(190_000_000, 1700);
+
+		let message = Xcm(vec![
+			Instruction::UnpaidExecution {
+				weight_limit: WeightLimit::Unlimited,
+				check_origin: None,
+			},
+			Instruction::Transact {
+				origin_kind: OriginKind::Native,
+				require_weight_at_most: call_weight,
+				call: request_core_count_call.encode().into(),
+			},
+		]);
+
+		match PolkadotXcm::send_xcm(Here, Location::parent(), message) {
+			Ok(_) => log::debug!(
+				target: "runtime::coretime",
+				"Request to update schedulable cores sent successfully."
+			),
+			Err(e) => log::error!(
+				target: "runtime::coretime",
+				"Failed to send request to update schedulable cores: {:?}",
+				e
+			),
+		}
+	}
+
+	fn request_revenue_info_at(when: RCBlockNumberOf<Self>) {
+		use crate::coretime::CoretimeProviderCalls::RequestRevenueInfoAt;
+		let _request_revenue_info_at_call =
+			RelayRuntimePallets::Coretime(RequestRevenueInfoAt(when));
+
+		log::debug!(
+			target: "runtime::coretime",
+			"`request_revenue` is unmiplemented on the relay."
+		);
+	}
+
+	fn credit_account(who: Self::AccountId, amount: Self::Balance) {
+		use crate::coretime::CoretimeProviderCalls::CreditAccount;
+		let _credit_account_call = RelayRuntimePallets::Coretime(CreditAccount(who, amount));
+
+		log::debug!(
+			target: "runtime::coretime",
+			"`credit_account` is unmiplemented on the relay."
+		);
+	}
+
+	fn assign_core(
+		core: CoreIndex,
+		begin: RCBlockNumberOf<Self>,
+		assignment: Vec<(CoreAssignment, PartsOf57600)>,
+		end_hint: Option<RCBlockNumberOf<Self>>,
+	) {
+		use crate::coretime::CoretimeProviderCalls::AssignCore;
+		let assign_core_call =
+			RelayRuntimePallets::Coretime(AssignCore(core, begin, assignment, end_hint));
+
+		// Weight for `assign_core` from Kusama runtime benchmarks:
+		// `ref_time` = 10177115 + (1 * 25000000) + (2 * 100000000) + (80 * 13932) = 236291675
+		// `proof_size` = 3612
+		// Add 5% to each component and round to 2 significant figures.
+		let call_weight = Weight::from_parts(248_000_000, 3800);
+
+		let message = Xcm(vec![
+			Instruction::UnpaidExecution {
+				weight_limit: WeightLimit::Unlimited,
+				check_origin: None,
+			},
+			Instruction::Transact {
+				origin_kind: OriginKind::Native,
+				require_weight_at_most: call_weight,
+				call: assign_core_call.encode().into(),
+			},
+		]);
+
+		match PolkadotXcm::send_xcm(Here, Location::parent(), message) {
+			Ok(_) => log::debug!(
+				target: "runtime::coretime",
+				"Core assignment sent successfully."
+			),
+			Err(e) => log::error!(
+				target: "runtime::coretime",
+				"Core assignment failed to send: {:?}",
+				e
+			),
+		}
+	}
+
+	fn check_notify_revenue_info() -> Option<(RCBlockNumberOf<Self>, Self::Balance)> {
+		let revenue = CoretimeRevenue::get();
```

Comment:
Suprised there is no `::take` on this

Reply:
It's defined with `parameter_types` macro which [doesn't seem to implement take](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/support/src/lib.rs#L336), only get and set (and key)